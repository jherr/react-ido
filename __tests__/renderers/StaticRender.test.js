import React from 'react';
import TestRenderer from 'react-test-renderer';

import {
  IdoSurface,
  StaticRender,
} from '../../src';

const Field = ({ name, value }) => (
  <div>
    {`${name}: ${value}`}
  </div>
);

const config = {
  widgets: [
    {
      type: 'field',
      name: 'Field',
      icon: 'list',
      component: Field,
      defaultValues: {
        name: 'Foo',
        value: 'Bar'
      }
    },
  ],
};

const elements = [
  [
    {
      type: 'field',
      id: 2,
      data: {
        name: 'Price',
        value: '$20.00',
      },
      width: 6,
    },
  ],
];

describe('StaticRenderer', () => {
  it('should render', () => {
    expect(
      TestRenderer.create(
        <IdoSurface
          config={config}
          elements={elements}
          onChange={() => {}}
        >
          <StaticRender />
        </IdoSurface>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
