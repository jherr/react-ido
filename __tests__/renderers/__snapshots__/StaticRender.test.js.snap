// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticRenderer should render 1`] = `
<div>
  <div
    className="css-1v0k9g2"
    style={
      Object {
        "gridTemplateColumns": "repeat(1, 1fr)",
      }
    }
  >
    <div>
      Price: $20.00
    </div>
  </div>
</div>
`;
